src/main/java/org/wildfly/security/auth/realm/ldap/LdapSecurityRealm.java

```diff
@@ -124,17 +125,16 @@ public ModifiableRealmIdentity getRealmIdentityForUpdate(String name, final Prin
     }
 
     @Override
-    public Iterator<ModifiableRealmIdentity> getRealmIdentityIterator() throws RealmUnavailableException {
+    public CloseableIterator<ModifiableRealmIdentity> getRealmIdentityIterator() throws RealmUnavailableException {
         if (identityMapping.iteratorFilter == null) {
             throw log.ldapRealmNotConfiguredToSupportIteratingOverIdentities();
         }
 
-        return new Iterator<ModifiableRealmIdentity>() {
+        return new CloseableIterator<ModifiableRealmIdentity>() {
```

Comment:
Shouldn't this have a close() implementation then?  Or does this impl not need one?